src/chains/tests/test_signals_ff_hook_events.py

```diff
@@ -0,0 +1,376 @@
+        assert len(responses.calls) == 3
+        assert isinstance(responses.calls[2], responses.Call)
+        assert responses.calls[
+            2
+        ].request.body == f'{{"type": "CHAIN_UPDATE", "chainId": "{chain.id}"}}'.encode(
+            "utf-8"
+        )
+        assert responses.calls[2].request.url == "http://127.0.0.1/v1/hooks/events"
+        assert (
+            responses.calls[2].request.headers.get("Authorization")
+            == "Basic example-token"
+        )
+
+    @responses.activate
+    def test_on_feature_delete_with_no_chain(self) -> None:
+        feature = Feature(key="Test Feature")
+
+        feature.save()  # create
+        feature.delete()  # delete
+
+        # Deleting a feature with no chains should not trigger any webhook
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    def test_on_feature_delete_with_chain(self) -> None:
+        chain = ChainFactory.create()
+        feature = FeatureFactory.create(key="Test Feature", chains=(chain,))
+
+        feature.delete()
+
+        # 1 call for Chain creation, 1 call for feature creation, 1 call for M2M update, 1 call for feature deletion
+        assert len(responses.calls) == 4
+        assert isinstance(responses.calls[3], responses.Call)
+        assert responses.calls[
+            3
+        ].request.body == f'{{"type": "CHAIN_UPDATE", "chainId": "{chain.id}"}}'.encode(
+            "utf-8"
+        )
+        assert responses.calls[3].request.url == "http://127.0.0.1/v1/hooks/events"
+        assert (
+            responses.calls[3].request.headers.get("Authorization")
+            == "Basic example-token"
+        )
+
+    @responses.activate
+    def test_on_feature_update_with_no_chain(self) -> None:
+        feature = Feature(key="Test Feature")
+
+        feature.save()  # create
+        feature.key = "New Test Feature"
+        feature.save()  # update
+
+        # Updating a feature with no chains should not trigger any webhook
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    def test_on_feature_update_with_chain(self) -> None:
+        chain = ChainFactory.create()
+        feature = FeatureFactory.create(key="Test Feature", chains=(chain,))
+
+        feature.chains.remove(chain)
+
+        # 1 call for Chain creation, 1 call for feature creation,
+        # 1 call for M2M update, 1 call for removing m2m relationship
+        assert len(responses.calls) == 4
+        assert isinstance(responses.calls[3], responses.Call)
+        assert responses.calls[
+            3
+        ].request.body == f'{{"type": "CHAIN_UPDATE", "chainId": "{chain.id}"}}'.encode(
+            "utf-8"
+        )
+        assert responses.calls[3].request.url == "http://127.0.0.1/v1/hooks/events"
+        assert (
+            responses.calls[3].request.headers.get("Authorization")
+            == "Basic example-token"
+        )
```

Comment:
Nitpick: do you think it would be useful to have a test for a feature update where the feature is associated with more than 1 chain? Including a second chain into one of these tests would suffice IMO. Just to test it iterates through chains when CRUDing features. 

This also applies to wallet-related webhooks tests.

Reply:
Done in [e4134f3](https://github.com/safe-global/safe-config-service/pull/964/commits/e4134f37a519ab64f57f25deaa388d49cd15a1f9)